src/main/java/com/example/recipebook5debag/service/impl/RecipeServiceImpl.java

```
@@ -0,0 +1,166 @@
+            throw new RuntimeException(e);
+        }
+    }
+
+    @PostConstruct
+    private void init() {
+        readDataFromFile();
+    }
+
+    private void readDataFromFile() {
+        try {
+            byte[] file = Files.readAllBytes(path);
+            Map<Long, Recipe> mapFromFile = objectMapper.readValue(file, new TypeReference<>() {
+            });
+            recipes.putAll(mapFromFile);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void writeDataToFile(Map<Long, Recipe> recipes) {
+        try {
+            byte[] bytes = objectMapper.writeValueAsBytes(recipes);
+            Files.write(path, bytes);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public Recipe addRecipe(Recipe recipe) {
+        if (recipes.containsKey(recipe.getId())) {
+            throw new RuntimeException("Такой рецепт уже добавлен");
+        } else {
+            recipes.put(lastId++, recipe);
+            writeDataToFile(recipes);
+        }
+        return recipe;
+    }
+
+
+    public Recipe getRecipeId(long id) {
+        if (recipes.containsKey(id)) {
+            return recipes.get(id);
+        } else {
+            throw new RuntimeException("Рецепт не найден");
+        }
+    }
+
+    public Collection<Recipe> getAllRecipes() {
+        return recipes.values();
+    }
+
+    public Recipe updateRecipe(long id, Recipe recipe) {
+        if (recipes.containsKey(id)) {
+            recipes.put(id, recipe);
+            writeDataToFile(recipes);
+            return recipe;
+        } else {
+            throw new RuntimeException("Рецепт не найден");
+        }
+    }
+
+    @Override
+    public Recipe remoweRecipe(long id) {
+        if (recipes.containsKey(id)) {
+            Recipe recipe = recipes.remove(id);
+            writeDataToFile(recipes);
+            return recipe;
+        }else {
+            throw new RuntimeException("Рецепт не найден");
```

все тоже самое, примени простой шабло
это называет return early / fail fast / ранний выход